File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/StudentResponseDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.AccountPrivacy;
import com.example.oriengo.model.enumeration.EducationLevel;
import com.example.oriengo.model.enumeration.GenderType;
import com.example.oriengo.model.enumeration.MessagePermission;
import com.example.oriengo.model.enumeration.VisibilityStatus;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StudentResponseDTO {

    private Long id;

    private String firstName;

    private String lastName;

    private Integer age;

    private GenderType gender;

    private String phoneNumber;

    private String email;

    private String school;

    private String fieldOfStudy;

    private EducationLevel educationLevel;

    private LocationDTO location;

    private VisibilityStatus profileVisibility;

    private MessagePermission messagePermission;

    private AccountPrivacy accountPrivacy;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

}

File: OrienGO-Frontend/src/app/model/dto/StudentResponse.dto.ts
import { AccountPrivacy } from "../enum/AccountPrivacy.enum";
import { EducationLevel } from "../enum/EducationLevel.enum";
import { MessagePermission } from "../enum/MessagePermission.enum";
import { VisibilityStatus } from "../enum/VisibilityStatus.enum";
import { GenderType } from "../enum/GenderType.enum";
import { LocationDTO } from "./Location.dto";

export interface StudentResponseDTO {
  id: number;

  firstName: string;
  lastName: string;
  age: number;
  gender: GenderType;

  phoneNumber: string;
  email: string;

  school?: string;
  fieldOfStudy?: string;
  educationLevel: EducationLevel;

  location?: LocationDTO;

  profileVisibility: VisibilityStatus;
  messagePermission: MessagePermission;
  accountPrivacy: AccountPrivacy;

  createdAt: string;  // LocalDateTime → ISO string
  updatedAt: string;
}
